lib/async_hooks.js

```diff
@@ -77,13 +77,13 @@ function fatalError(e) {
 
 class AsyncHook {
   constructor({ init, before, after, destroy }) {
-    if (init && typeof init !== 'function')
+    if (typeof init !== 'undefined' && typeof init !== 'function')
```

Comment:
Because without them it will accept `false` without throwing, which I assume is not the expected behaviour